### PR TITLE
Add TAP reporter

### PR DESCRIPTION
**Pull request by frostyfrog**
This is a hackish attempt to add a TAP reporter (see
philsquared/Catch#309 ) by following the TAP 12 specification
http://testanything.org/tap-specification.html. I'm unsure how well I
did in following the spec or with following good C++ guidelines.
Comments are appreciated.

Signed-off-by: Colton Wolkins (Ogre) frostyfrog2@gmail.com

Sending a pull request because someone asked. I didn't realize that it had gotten 63 commits behind. >_> I also don't think that I really need to be in the "created by" line since I mostly copied from another reported than modified it until I got it to somewhat work. Sorry! :crying_cat_face:

### CODE REVIEW

**Comment by Qix-** [comment]
*poke* could this be reviewed please? This would be awesome to see.

File: include/reporters/catch_reporter_tap.hpp [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+                }
+            }
+
+            void printReconstructedExpression() const {
+                if( result.hasExpandedExpression() ) {
+                    {
+                        Colour colour( dimColour() );
+                        stream << " for: ";
+                    }
+										std::string expr = result.getExpandedExpression();
```
Comment (Bu11etmagnet) [inline include/reporters/catch_reporter_tap.hpp]: What's up with the indentation? Did you use tabs and expect them to be four spaces?

**Comment by horenmar** [comment]
I finally got around to this, will need some time to cherry-pick and merge it.

**Comment by horenmar** [comment]
This should be now part of the master.

**Pull request closed without merging**